lib/engine/round/operating.rb

```diff
@@ -508,16 +523,17 @@ def buy_company(company, price)
       def place_token(action)
         entity = action.entity
         hex = action.city.hex
-        if !@game.loading && !connected_nodes[action.city] && !@teleported
+        free = @teleported || @step == :home_token
+        if !@game.loading && !connected_nodes[action.city] && !free
```

Comment:
move the free check before connected_nodes because connected nodes is an expensive call and better to short circuit

Reply:
hmmm but free refers to cost but you're overloading it here with connectivity?